### PR TITLE
Put 4844 rules in serenity consensus engine

### PR DESCRIPTION
**Pull request by roberto-bayardo**
Also:

- make sure Withdrawals field is never nil after decoding post-capella encoded blocks & bodies (this is a bug that I should also submit upstream)

- add test for encoding/decoding of excess data gas field

- go mod tidy

### CODE REVIEW

**Comment by roberto-bayardo** [comment]
ping @0x00101010

File: core/types/block.go [hunk 1]
```diff
@@ -516,6 +516,7 @@ func (h *Header) SetExcessDataGas(v *big.Int) {
 		h.ExcessDataGas.Set(v)
 	}
 	if h.WithdrawalsHash == nil {
+		fmt.Println("Setting withdrawals hash to empty root hash")
```
Comment (0x00101010) [inline core/types/block.go:519]: curious, no logging library is used and we use `fmt.Println` for logging?

Comment (0x00101010) [reply core/types/block.go:519]: I see log / fmt.Println both being used 🤔

**Review by 0x00101010 — APPROVED** [review]
nit question about logging

File: core/types/block.go (hunk 1 above)
Comment (roberto-bayardo) [reply core/types/block.go:519]: Good catch I meant to remove this (it's from my debugging.) I will remove it.

**Pull request merged**